cpp/oneapi/dal/test/engine/config.cpp

```diff
@@ -65,7 +67,14 @@ void global_cleanup() {
 }
 
 #else
-void global_setup(const global_config& config) {}
+void global_setup(const global_config& config) {
+    if (config.device_selector != "" && config.device_selector != "cpu") {
+        throw std::invalid_argument{
+            "Test is build in HOST mode, so only CPU device is available"
+        };
+    }
+}
```

Comment:
shouldn't we use own exceptions?

Reply:
It's better to throw std exceptions from the test engine, because they can be distinguished from the dal exceptions in particular test.